gravitee-exchange-controller/gravitee-exchange-controller-core/src/main/java/io/gravitee/exchange/controller/core/cluster/ControllerClusterManager.java

```diff
@@ -99,6 +120,42 @@ private void handleClusteredReply(Message<ClusteredReply<?>> clusteredReplyMessa
         }
     }
 
+    private void handleChannelsRebalancing() {
+        log.debug("[{}] New cluster member detected, starting re-balancing checks", identifyConfiguration.id());
+
+        int clusterSize = clusterManager.members().size();
+        if (clusterSize > 1) {
+            targetsMetric()
+                // Keep only target with more than 1 channel connected
+                .filter(targetMetric -> targetMetric.channelMetrics().size() > 1)
+                .flatMapIterable(TargetMetric::channelMetrics)
+                .toList()
+                // Reduce the list of 1/members elements
+                .flattenStreamAsFlowable(channelMetrics -> {
+                    int size = channelMetrics.size();
+                    int reducedSize = Math.round((float) size / clusterSize);
+                    log.debug("[{}] {} channels will be scheduled for re-balancing", identifyConfiguration.id(), reducedSize);
+                    return channelMetrics.stream().limit(reducedSize);
+                })
+                .map(channelMetric -> channelManager.getChannelById(channelMetric.id()))
+                .filter(Objects::nonNull)
+                // Filter channel with pending commands
+                .filter(controllerChannel -> !controllerChannel.hasPendingCommands())
+                // Unregister channel (close with reconnection)
+                .flatMapCompletable(controllerChannel -> {
+                    log.debug(
+                        "[{}] Re-balancing channel [{}] for the target [{}]",
+                        identifyConfiguration.id(),
+                        controllerChannel.id(),
+                        controllerChannel.targetId()
+                    );
+                    return unregister(controllerChannel).onErrorComplete();
```

Comment:
What happens on the client side? Does it reconnect immediately or is there any delay applied?